packages/jest-cli/src/cli/index.js

```diff
@@ -34,33 +34,35 @@ import watch from '../watch';
 import yargs from 'yargs';
 
 export async function run(maybeArgv?: Argv, project?: Path) {
-  const argv: Argv = _buildArgv(maybeArgv, project);
-  const projects = _getProjectListFromCLIArgs(argv, project);
+  const argv: Argv = buildArgv(maybeArgv, project);
+  const projects = getProjectListFromCLIArgs(argv, project);
   // If we're running a single Jest project, we might want to use another
   // version of Jest (the one that is specified in this project's package.json)
-  const runCLIFn = _getRunCLIFn(projects);
+  const runCLIFn = getRunCLIFn(projects);
 
   const {results, globalConfig} = await runCLIFn(argv, projects);
-  _readResultsAndExit(results, globalConfig);
+  readResultsAndExit(results, globalConfig);
 }
 
 export const runCLI = async (
   argv: Argv,
   projects: Array<Path>,
 ): Promise<{results: AggregatedResult, globalConfig: GlobalConfig}> => {
+  const realFs = require('fs');
+  const fs = require('graceful-fs');
+  fs.gracefulify(realFs);
```

Comment:
Isn't there an issue if this happens multiple times?

https://github.com/isaacs/node-graceful-fs/#changes

>There are advantages to each approach.  Monkey-patching the fs means
that no `EMFILE` errors can possibly occur anywhere in your
application, because everything is using the same core `fs` module,
which is patched.  However, it can also obviously cause undesirable
side-effects, **especially if the module is loaded multiple times.**

(emphasis mine)